src/entities/swap.ts

```diff
@@ -206,6 +207,145 @@ export class Swap {
+        }
+
+        if (!this.isBatchSwap) {
+            return [limitAmount];
+        }
+
+        for (let i = 0; i < this.assets.length; i++) {
+            if (
+                this.assets[i] === this.inputAmount.token.address ||
+                (this.assets[i] === ZERO_ADDRESS &&
+                    this.inputAmount.token.address === NATIVE_ADDRESS)
+            ) {
+                if (this.swapKind === SwapKind.GivenIn) {
+                    limits[i] = this.inputAmount.amount;
+                } else {
+                    limits[i] = limitAmount;
+                }
+            }
+            if (
+                this.assets[i] === this.outputAmount.token.address ||
+                (this.assets[i] === ZERO_ADDRESS &&
+                    this.outputAmount.token.address === NATIVE_ADDRESS)
+            ) {
+                if (this.swapKind === SwapKind.GivenIn) {
+                    limits[i] = -1n * limitAmount;
+                } else {
+                    limits[i] = -1n * this.outputAmount.amount;
+                }
+            }
+        }
+
+        return limits;
+    }
+
+    /**
+     * Returns the transaction data to be sent to the vault contract
+     *
+     * @param limits calculated from swap.limits()
+     * @param deadline unix timestamp
+     * @param sender address of the sender
+     * @param recipient defaults to sender
+     * @returns
+     */
+    transactionData(
+        limits: bigint[],
+        deadline: bigint,
+        sender: Address,
+        recipient = sender,
+    ) {
+        return {
+            to: this.to(),
+            data: this.callData(limits, deadline, sender, recipient),
+            value: this.value(limits),
+        };
+    }
+
+    /**
+     * Returns the native assset value to be sent to the vault contract based on the swap kind and the limit amounts
+     *
+     * @param limits calculated from swap.limits()
+     * @returns
+     */
+    private value(limits: bigint[]): bigint {
+        let value = 0n;
+        if (this.inputAmount.token.address === NATIVE_ADDRESS) {
+            const idx = this.assets.indexOf(ZERO_ADDRESS);
+            value = limits[idx];
+        }
+        return value;
+    }
+
+    private to(): Address {
+        return VAULT[this.chainId];
+    }
+
+    /**
+     * Returns the call data to be sent to the vault contract for the swap execution.
+     *
+     * @param limits calculated from swap.limits()
+     * @param deadline unix timestamp
+     * @param sender address of the sender
+     * @param recipient defaults to sender
+     * @returns
+     */
+    private callData(
+        limits: bigint[],
+        deadline: bigint,
+        sender: Address,
+        recipient = sender,
+    ): Hex {
+        let callData: Hex;
+
+        const funds = {
+            sender,
+            recipient,
+            fromInternalBalance: false,
+            toInternalBalance: false,
```

Comment:
Shouldn't we allow these to be set by the user?

Reply:
true, i'll set them to false by default. no one seems to be using these.